diff --git a/Makefile.am b/Makefile.am
index 55cae89fa..8ce1af3b8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -9,13 +9,13 @@ if CONFIG_USER
 SUBDIRS += udev etc man scripts lib tests cmd contrib
 endif
 if CONFIG_KERNEL
-SUBDIRS += module
+SUBDIRS += scripts module
 
 extradir = $(prefix)/src/zfs-$(VERSION)
-extra_HEADERS = zfs.release.in zfs_config.h.in
+extra_HEADERS = zfs.release zfs_config.h
 
 kerneldir = $(prefix)/src/zfs-$(VERSION)/$(LINUX_VERSION)
-nodist_kernel_HEADERS = zfs.release zfs_config.h module/$(LINUX_SYMBOLS)
+nodist_kernel_HEADERS = zfs.release.in zfs_config.h.in module/$(LINUX_SYMBOLS)
 endif
 
 AUTOMAKE_OPTIONS = foreign
@@ -54,8 +54,8 @@ install-data-hook:
 	cd $(DESTDIR)$(prefix)/src/spl-$(VERSION) && \
 	ln -s ../zfs-$(VERSION)/include/spl include && \
 	ln -s ../zfs-$(VERSION)/$(LINUX_VERSION) $(LINUX_VERSION) && \
-	ln -s ../zfs-$(VERSION)/zfs_config.h.in spl_config.h.in && \
-	ln -s ../zfs-$(VERSION)/zfs.release.in spl.release.in && \
+	ln -s ../zfs-$(VERSION)/zfs_config.h spl_config.h && \
+	ln -s ../zfs-$(VERSION)/zfs.release spl.release && \
 	cd $(DESTDIR)$(prefix)/src/zfs-$(VERSION)/$(LINUX_VERSION) && \
 	ln -fs zfs_config.h spl_config.h && \
 	ln -fs zfs.release spl.release
