--- zfs-zfs-0.8.3/lib/libnvpair/Makefile.am~	2020-01-22 22:51:17.000000000 +0100
+++ zfs-zfs-0.8.3/lib/libnvpair/Makefile.am	2020-04-05 10:21:40.550628294 +0200
@@ -28,7 +28,9 @@
 	$(USER_C) \
 	$(KERNEL_C)
 
-libnvpair_la_LIBADD = $(LIBTIRPC_LIBS)
+libnvpair_la_LIBADD = $(LIBTIRPC_LIBS) \
+	$(top_builddir)/lib/libspl/libspl.la
+
 libnvpair_la_LDFLAGS = -version-info 1:1:0
 
 EXTRA_DIST = $(USER_C)
