--- zfs-2.3.1/config/zfs-build.m4.orig	2025-04-15 17:48:26.176529494 +0200
+++ zfs-2.3.1/config/zfs-build.m4	2025-04-15 18:15:07.527854229 +0200
@@ -629,7 +629,7 @@ AC_DEFUN([ZFS_AC_DEFAULT_PACKAGE], [
 
 	AC_MSG_CHECKING([default bash completion directory])
 	case "$VENDOR" in
-		alpine|arch|artix|debian|gentoo|ubuntu)
+		alpine|arch|artix|debian|gentoo|ubuntu|pld)
 			bashcompletiondir=/usr/share/bash-completion/completions
 			;;
 		freebsd)
