--- zfs-0.7.3/tests/zfs-tests/cmd/xattrtest/xattrtest.c~	2017-10-18 23:43:13.000000000 +0200
+++ zfs-0.7.3/tests/zfs-tests/cmd/xattrtest/xattrtest.c	2017-11-18 17:15:18.389310337 +0100
@@ -37,7 +37,7 @@
 #include <fcntl.h>
 #include <time.h>
 #include <unistd.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
--- zfs-0.7.3/tests/zfs-tests/cmd/mktree/mktree.c~	2017-10-18 23:43:13.000000000 +0200
+++ zfs-0.7.3/tests/zfs-tests/cmd/mktree/mktree.c	2017-11-18 17:15:46.816754985 +0100
@@ -30,7 +30,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <fcntl.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/param.h>
--- zfs-0.7.3/config/user-libattr.m4	2017-10-18 23:43:14.000000000 +0200
+++ /dev/null	2017-11-04 00:31:38.176382048 +0100
@@ -1,12 +0,0 @@
-dnl #
-dnl # Check for libattr
-dnl #
-AC_DEFUN([ZFS_AC_CONFIG_USER_LIBATTR], [
-	LIBATTR=
-
-	AC_CHECK_HEADER([attr/xattr.h], [], [AC_MSG_FAILURE([
-	*** attr/xattr.h missing, libattr-devel package required])])
-
-	AC_SUBST([LIBATTR], ["-lattr"])
-	AC_DEFINE([HAVE_LIBATTR], 1, [Define if you have libattr])
-])
--- zfs-0.7.3/config/user.m4~	2017-10-18 23:43:14.000000000 +0200
+++ zfs-0.7.3/config/user.m4	2017-11-18 17:17:18.399261345 +0100
@@ -11,7 +11,6 @@
 	ZFS_AC_CONFIG_USER_LIBUUID
 	ZFS_AC_CONFIG_USER_LIBTIRPC
 	ZFS_AC_CONFIG_USER_LIBBLKID
-	ZFS_AC_CONFIG_USER_LIBATTR
 	ZFS_AC_CONFIG_USER_LIBUDEV
 	ZFS_AC_CONFIG_USER_FRAME_LARGER_THAN
 	ZFS_AC_CONFIG_USER_RUNSTATEDIR
