--- zfs-zfs-0.8.0-rc4/config/always-python.m4.orig	2019-04-16 22:24:49.000000000 +0200
+++ zfs-zfs-0.8.0-rc4/config/always-python.m4	2019-05-08 18:38:16.579298208 +0200
@@ -68,19 +68,19 @@
 
 	AM_PATH_PYTHON([2.6], [], [:])
 	AM_CONDITIONAL([USING_PYTHON], [test "$PYTHON" != :])
-	AM_CONDITIONAL([USING_PYTHON_2], [test "${PYTHON_VERSION:0:2}" = "2."])
-	AM_CONDITIONAL([USING_PYTHON_3], [test "${PYTHON_VERSION:0:2}" = "3."])
+	AM_CONDITIONAL([USING_PYTHON_2], [test "${PYTHON_VERSION%%.*}" = "2"])
+	AM_CONDITIONAL([USING_PYTHON_3], [test "${PYTHON_VERSION%%.*}" = "3"])
 
 	dnl #
 	dnl # Minimum supported Python versions for utilities:
 	dnl # Python 2.6.x, or Python 3.4.x
 	dnl #
-	AS_IF([test "${PYTHON_VERSION:0:2}" = "2."], [
+	AS_IF([test "${PYTHON_VERSION%%.*}" = "2"], [
 		ZFS_AC_PYTHON_VERSION([>= '2.6'], [ /bin/true ],
 			[AC_MSG_ERROR("Python >= 2.6.x is not available")])
 	])
 
-	AS_IF([test "${PYTHON_VERSION:0:2}" = "3."], [
+	AS_IF([test "${PYTHON_VERSION%%.*}" = "3"], [
 		ZFS_AC_PYTHON_VERSION([>= '3.4'], [ /bin/true ],
 			[AC_MSG_ERROR("Python >= 3.4.x is not available")])
 	])
--- zfs-zfs-0.8.0-rc4/config/always-pyzfs.m4.orig	2019-04-16 22:24:49.000000000 +0200
+++ zfs-zfs-0.8.0-rc4/config/always-pyzfs.m4	2019-05-08 19:29:53.339188287 +0200
@@ -26,10 +26,10 @@
 	dnl # Require python-devel libraries
 	dnl #
 	AS_IF([test "x$enable_pyzfs" = xcheck  -o "x$enable_pyzfs" = xyes], [
-		AS_IF([test "${PYTHON_VERSION:0:2}" = "2."], [
+		AS_IF([test "${PYTHON_VERSION%%.*}" = "2"], [
 			PYTHON_REQUIRED_VERSION=">= '2.7.0'"
 		], [
-			AS_IF([test "${PYTHON_VERSION:0:2}" = "3."], [
+			AS_IF([test "${PYTHON_VERSION%%.*}" = "3"], [
 				PYTHON_REQUIRED_VERSION=">= '3.4.0'"
 			], [
 				AC_MSG_ERROR("Python $PYTHON_VERSION unknown")
